nemo_aligner/utils/trt_llm.py

```diff
@@ -55,32 +70,44 @@ def __init__(
         self.unload_engine_train = unload_engine_train
         self.trt_model_type = trt_model_type
         self.reshard_model = reshard_model
-        self.pad_id = tokenizer.eos_id
-
         self.trt_llm_exporter = TensorRTLLM(trt_model_dir, load_model=False)
         self._trtllm_model_compiled = False
 
         rng_generator = torch.Generator(device="cpu")
         seed = secrets.randbits(32) if seed is None else seed
         rng_generator.manual_seed(seed)
-
         self.rng_generator = rng_generator
 
+        self.pad_id = tokenizer.pad_id
+        end_id = tokenizer.eos_id
         end_strings = list(end_strings)
-        assert all(
-            "," not in end_string for end_string in end_strings
-        ), "detected `,` in a specified end_string. This is will cause an error when converting it into ids for TRT-LLM"
 
-        end_strings = [[",".join(end_strings)] for _ in range(self.generation_batch_size)]
+        # TRT-LLM uses different logic to compute the response length depending on if we specify stop_list or end_id
+        # we want to use the same logic(which is to include the stop token in response length) so we put end_id into the stop_list
+        # manually
+        if len(end_strings) == 0:
+            ids, offsets = [end_id], [1]
+        else:
+            assert all(
+                "," not in end_string for end_string in end_strings
+            ), "detected `,` in a specified end_string. This is will cause an error when converting it into ids for TRT-LLM"
+
+            end_strings = [[",".join(end_strings)] for _ in range(self.generation_batch_size)]
```

Comment:
Seems to me that we only need batch size 1 here since we're only using the first element then repeating the result.
```suggestion
            end_strings = [[",".join(end_strings)]]
```

Reply:
sure! done 1b638322630aa5ddf02019e5ee0a7a93a7b528e2